src/ExternalEvents.php

```diff
@@ -41,6 +44,11 @@ public static function decorateListener(string $listenerClass): \Closure
             try {
                 $listener = resolve($listenerClass);
 
+                if (!method_exists($listener, 'setClient')) {
+                    throw new ReflectionException();
```

Comment:
You can modify the catch to capture two exceptions instead of force here an exception that is not real.